src/Sylius/Component/Addressing/Model/Province.php

```diff
@@ -112,4 +112,20 @@ public function setCountry(CountryInterface $country = null)
 
         return $this;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isEqualTo($data)
+    {
+        if (is_object($data)) {
+            if (!$data instanceof ProvinceInterface) {
+                return false;
+            }
+
+            return $data->getId() === $this->id;
+        }
+
+        return $data === $this->id;
```

Comment:
This is done this way to handle BC when we have just IDs of entities instead of whole entity (think promotions etc.)